### PR TITLE
Added functional Sheridan College scraper

### PR DESCRIPTION
**Pull request by HarrisonGregg**

### CODE REVIEW

File: osp_scraper/spiders/sheridanc.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    name = option.css("option::text").extract_first()
+                    seasons.append((code, name))
+
+            for option in response.css("#program_ps option"):
+                code = option.css("option::attr(value)").extract_first()
+                if code:
+                    name = option.css("option::text").extract_first()
+                    programs.append((code, name))
+
+            for year in range(2000, 2020):
```
Comment (jodizzle) [inline osp_scraper/spiders/sheridanc.py]: Is there a particular reason you cut it off at 2020?

Funnily enough, I was playing around with the website, and if you have javascript turned off there's no popup window that says there are only outlines "for Fall 2000 and later", allowing you to enter searches for earlier terms.  I don't think this really affects your scraper though, as earlier terms don't seem very content-full.

File: osp_scraper/spiders/sheridanc.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                                "season": season_code,
+                                "year": str(year),
+                                "Search": "Search"
+                            },
+                            callback=self.parse_results
+                        )
+
+        yield scrapy.Request(start_url, callback=get_parameters)
+
+    def parse_results(self, response):
```
Comment (jodizzle) [inline osp_scraper/spiders/sheridanc.py]: Is there a reason why this function isn't nested?  Nesting functions up to the point of `callback=self.parse_for_files` seems to have been the go-to style for the scrapers so far.

File: osp_scraper/spiders/sheridanc.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        for row in response.css('#detail_table tr'):
+            relative_url = row.css('a::attr(href)').extract_first()
+            if relative_url:
+                url = response.urljoin(relative_url)
+                anchor = self.clean_whitespace(row.css('tr::text').extract_first())
+                yield scrapy.Request(
+                    url,
```
Comment (jodizzle) [inline osp_scraper/spiders/sheridanc.py]: Minor: Trailing whitespace.

File: osp_scraper/spiders/sheridanc.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                    url, 
+                    meta={
+                        'depth': 1,
+                        'hops_from_seed': 1,
+                        'source_url': response.url,
+                        'source_anchor': anchor
+                    },
+                    callback=self.parse_for_files
+                )
+
```
Comment (jodizzle) [inline osp_scraper/spiders/sheridanc.py]: Minor: Unnecessary whitespace.

File: osp_scraper/spiders/sheridanc.py (hunk 1 above)
Comment (HarrisonGregg) [inline osp_scraper/spiders/sheridanc.py]: I just chose 2020 arbitrarily.  Figured I'd give it a bit of buffer into the future but didn't put too much thought into it.

File: osp_scraper/spiders/sheridanc.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            for option in response.css("#program_ps option"):
+                code = option.css("option::attr(value)").extract_first()
+                if code:
+                    name = option.css("option::text").extract_first()
+                    programs.append((code, name))
+
+            for year in range(2000, 2020):
+                for season_code, season_name in seasons:
+                    for program_code, program_name in programs:
```
Comment (HarrisonGregg) [inline osp_scraper/spiders/sheridanc.py]: I've refactored it to clean it up a little.

File: osp_scraper/spiders/sheridanc.py (hunk 2 above)
Comment (HarrisonGregg) [inline osp_scraper/spiders/sheridanc.py]: My original logic in writing these was to nest functions that would only be called once, and to put functions called repeatedly on the class itself, but I realize I haven't been following this consistently.

**Comment by jodizzle** [comment]
I think my question about the years range is still open, but it's mostly a curiosity and not that important.

OK to merge.

**Comment by HarrisonGregg** [comment]
@jodizzle I decided to determine the year dynamically and I refactored a little.  Still look good to go?

**Comment by jodizzle** [comment]
Looks good.  Just be cautious about running on computers with very out of date clocks.

**Comment by HarrisonGregg** [comment]
@jodizzle Challenge accepted.

**Comment by jodizzle** [comment]
Nice, you did me one better.  OK to merge.
